src/ucs/arch/x86_64/cpu.c

```diff
@@ -12,22 +12,46 @@
 #include <ucs/sys/sys.h>
 #include <ucs/sys/string.h>
 
+#define X86_CPUID_GENUINEINTEL    "GenuntelineI" /* GenuineIntel in magic notation */
+#define X86_CPUID_AUTHENTICAMD    "AuthcAMDenti" /* AuthenticAMD in magic notation */
 #define X86_CPUID_GET_MODEL       0x00000001u
 #define X86_CPUID_GET_BASE_VALUE  0x00000000u
 #define X86_CPUID_GET_EXTD_VALUE  0x00000007u
 #define X86_CPUID_GET_MAX_VALUE   0x80000000u
 #define X86_CPUID_INVARIANT_TSC   0x80000007u
 
 
+typedef union x86_cpu_registers {
+    struct {
+        union {
+            uint32_t     eax;
+            uint8_t      max_iter; /* leaf 2 - max iterations */
+        };
+        union {
+            struct {
+                uint32_t ebx;
+                uint32_t ecx;
+                uint32_t edx;
+            };
+            char         id[sizeof(uint32_t) * 3]; /* leaf 0 - CPU ID */
```

Comment:
maybe use `uint8_t` as well?

Reply:
actually it is string, char here used to simplify debugging